docs/proxy.md

```diff
@@ -0,0 +1,128 @@
+verify return:1
+depth=0 CN = github.com
+verify error:num=21:unable to verify the first certificate
+verify return:1
+depth=0 CN = github.com
+verify return:1
+CONNECTED(00000003)
+---
+Certificate chain
+ 0 s:CN = github.com
+   i:CN = mitmproxy, O = mitmproxy
+   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
+   v:NotBefore: Nov  7 15:43:48 2024 GMT; NotAfter: Nov  9 15:43:48 2025 GMT
+---
+Server certificate
+-----BEGIN CERTIFICATE-----
+```
+
+Let's rerun Envbuilder using the proxy to see how it responds. To do this, we use the same command as before, except that we also set the `https_proxy` environment variable:
+```bash
+docker run -it --rm \
+    -e https_proxy=https://172.17.0.2:8080 \
+    -e ENVBUILDER_INIT_SCRIPT='/bin/sh' \
+    -e ENVBUILDER_GIT_URL='https://github.com/coder/envbuilder.git' \
+    ghcr.io/coder/envbuilder:latest
+```
+From the logs, notice that certificate verification fails. It fails because it doesn't trust the certificate that was provided by mitmproxy:
+```
+Failed to clone repository: clone "https://github.com/coder/envbuilder.git": Get "https://github.com/coder/envbuilder.git/info/refs?service=git-upload-pack": proxyconnect tcp: tls: failed to verify certificate: x509: certificate signed by unknown authority
+```
+
+To fix this, we need to provide a ca certificate that Envbuilder can use to verify the server certificate that mitmproxy serves instead of github's actual certificate. Envbuilder provides a few environment variables to accomplish this. They are documented in the summary section above. For this example, we have the ca certificate saved in a directory. The easiest way to provide the ca certificate to envbuilder is therefore to mount it as a volume in the envbuilder container and tell envbuilder to use it. For this, we can use the `SSL_CERT_FILE` environment variable. The command to run Envbuilder is now:
+```bash
+docker run -it --rm \
+    -v $PWD/certs:/certs \
+    -e SSL_CERT_FILE=/certs/mitmproxy-ca-cert.pem \
+    -e https_proxy=https://172.17.0.2:8080 \
+    -e ENVBUILDER_INIT_SCRIPT='/bin/sh' \
+    -e ENVBUILDER_GIT_URL='https://github.com/coder/envbuilder.git' \
+    ghcr.io/coder/envbuilder:latest
+```
+
+As before, this command should yield a shell inside an Envbuilder built environment. Feel free to test it en then exit the container. Assuming this worked, Envbuilder will have cloned a repository and built the relevant container using a proxy that required a custom ca certificate.
+
+### Bonus
+To prove that Envbuilder did in fact use the proxy, and also because it is interesting to observe, open `http://localhost:8081/` in your local browser and you should see the mitmproxy web interface. In the flow tab, there should be a list of all of the HTTP requests that were required to build the container. The first three will be the requests that were used by git to clone the repository. The rest will be the requests that were used to clone the devcontainer image.
```

Comment:
```suggestion
To prove that Envbuilder did in fact use the proxy, and also because it is interesting to observe, open `http://localhost:8081/` in your local browser and you should see the mitmproxy web interface. In the flow tab, there should be a list of all of the HTTP requests that were required to build the container. The first three will be the requests that were used to clone the Git repository. The rest will be the requests that were used to pull the devcontainer image.
```
Nit: There's still a bit of uncertainty here (should) and certainty that the Git operations will only perform three requests (I feel this could change).

Reply:
updated